src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/ListViewModel.cs

```diff
@@ -18,20 +19,45 @@ public partial class ListViewModel : ObservableObject
     [ObservableProperty]
     public partial ObservableGroupedCollection<string, ListItemViewModel> Items { get; set; } = [];
 
+    private readonly ExtensionObject<IListPage> _model;
+
+    private readonly TaskScheduler _taskScheduler;
+
     public ListViewModel(IListPage model)
     {
-        // TEMPORARY: just plop all the items into a single group
-        // see 9806fe5d8 for the last commit that had this with sections
-        ObservableGroup<string, ListItemViewModel> group = new(string.Empty);
-
-        foreach (var item in model.GetItems())
-        {
-            ListItemViewModel viewModel = new(item);
-            _ = viewModel.InitializePropertiesAsync();
-            group.Add(viewModel);
-        }
-
-        Items.AddGroup(group);
+        _model = new(model);
+        _taskScheduler = TaskScheduler.FromCurrentSynchronizationContext();
+
+        FetchItems();
+        model.ItemsChanged += Model_ItemsChanged;
+    }
+
+    private void Model_ItemsChanged(object sender, ItemsChangedEventArgs args) => FetchItems();
+
+    private void FetchItems()
+    {
+        Task.Factory.StartNew(
+           () =>
+       {
+           // TEMPORARY: just plop all the items into a single group
+           // see 9806fe5d8 for the last commit that had this with sections
+           ObservableGroup<string, ListItemViewModel> group = new(string.Empty);
+
+           // TODO unsafe
+           var newItems = _model.Unsafe!.GetItems();
+           Items.Clear();
+           foreach (var item in newItems)
+           {
+               ListItemViewModel viewModel = new(item);
+               _ = viewModel.InitializePropertiesAsync();
+               group.Add(viewModel);
```

Comment:
I would imagine we only have to dispatch the add to the ObservableCollection here? Vs. the whole operation. I guess we haven't hit an issue with the built-in extensions due to them being in the same proc?

#180 though abstracts the Loading of the ViewModel away to the async background thread for us, so then you don't have to do the `Wait` there for the fetch.